bridge-worker/server_test.go

```diff
@@ -0,0 +1,351 @@
+			setupRequest: func() *types.SubmitSignatureRequest {
+				entry := createValidEntry()
+				entry.Amount = nil
+				signature := "0x" + strings.Repeat("00", 65)
+				return &types.SubmitSignatureRequest{
+					Entry:     entry,
+					Signature: signature,
+				}
+			},
+		},
+		{
+			name:           "Invalid amount (zero)",
+			expectedStatus: http.StatusBadRequest,
+			expectedError:  "invalid amount",
+			setupRequest: func() *types.SubmitSignatureRequest {
+				entry := createValidEntry()
+				entry.Amount = big.NewInt(0)
+				signature := "0x" + strings.Repeat("00", 65)
+				return &types.SubmitSignatureRequest{
+					Entry:     entry,
+					Signature: signature,
+				}
+			},
+		},
+		{
+			name:           "Invalid amount (negative)",
+			expectedStatus: http.StatusBadRequest,
+			expectedError:  "invalid amount",
+			setupRequest: func() *types.SubmitSignatureRequest {
+				entry := createValidEntry()
+				entry.Amount = big.NewInt(-1)
+				signature := "0x" + strings.Repeat("00", 65)
+				return &types.SubmitSignatureRequest{
+					Entry:     entry,
+					Signature: signature,
+				}
+			},
+		},
+		{
+			name:           "Invalid chain (out of range)",
+			expectedStatus: http.StatusBadRequest,
+			expectedError:  "invalid chain",
+			setupRequest: func() *types.SubmitSignatureRequest {
+				entry := createValidEntry()
+				entry.Chain = 99 // Invalid chain value
+				signature := "0x" + strings.Repeat("00", 65)
+				return &types.SubmitSignatureRequest{
+					Entry:     entry,
+					Signature: signature,
+				}
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			var requestBody []byte
+			var err error
+
+			if testCase.setupRequest != nil {
+				req := testCase.setupRequest()
+				requestBody, err = json.Marshal(req)
+				require.NoError(t, err)
+			} else if testCase.requestBody != nil {
+				if str, ok := testCase.requestBody.(string); ok {
+					requestBody = []byte(str)
+				} else {
+					requestBody, err = json.Marshal(testCase.requestBody)
+					require.NoError(t, err)
+				}
+			}
+
+			req := httptest.NewRequest(http.MethodPost, "/submit-signature", bytes.NewReader(requestBody))
+			req.Header.Set("Content-Type", "application/json")
+
+			w := httptest.NewRecorder()
+
+			server.submitSignature(w, req)
+
+			assert.Equal(t, testCase.expectedStatus, w.Code)
+			assert.Equal(t, "application/json", w.Header().Get("Content-Type"))
+
+			var response types.SubmitSignatureResponse
+			err = json.Unmarshal(w.Body.Bytes(), &response)
+			require.NoError(t, err)
+
+			// Check response content
+			if testCase.expectedStatus == http.StatusAccepted {
+				assert.True(t, response.Success)
+				assert.Empty(t, response.Error)
+			} else {
+				assert.Equal(t, w.Code, testCase.expectedStatus)
+				assert.False(t, response.Success)
+				assert.Contains(t, response.Error, testCase.expectedError)
+			}
+		})
+	}
+}
+
+func TestServer_recoverAddress(t *testing.T) {
```

Comment:
I believe we should provide proper test vectors for this function and treat the tested function as a black box. This test must ensure the signature will be accepted by the contract. Right now, we are just taking the logic of the tested function, and reverse it to produce the test vector. This will always pass. See that we are missing a keccak256 that is checked on the contract side and this test still pass. Let's use existing Solidity tests to get a test vector for this test: https://github.com/thesis/mezo-portal/blob/47fcf2dd33c681c9ef41df2b8ff470b5ef4fab47/solidity/test/MezoBridge.test.ts#L209